9.tcga-classify/scripts/nbconverted/interpret-all-feature-ensemble-model.py

```diff
@@ -16,128 +16,165 @@
 import pandas as pd
 
 
-# ## Load the Top Model Identified Previously
+# ## Load the `All Feature` Ensemble Model
 
 # In[2]:
 
 
-model_file = os.path.join("results", "top_model_ensemble_tp53_feature_for_followup.tsv")
+model_file = os.path.join("results", "top_model_ensemble_all_features_tp53_feature_for_followup.tsv")
 top_model_df = pd.read_table(model_file)
 top_model_df
 
 
 # In[3]:
 
 
-# The seed we used to compile single model
-seed = "165158"
-z_dim = top_model_df.z_dim.values[0]
+coef_file = os.path.join("results",
+                         "mutation_ensemble_all",
+                         "TP53",
+                         "TP53_ensemble_all_features_coefficients.tsv.gz")
+coef_df = pd.read_table(coef_file).drop(['signal', 'z_dim', 'seed', 'algorithm'], axis='columns')
+coef_df.head()
 
 
-# ## Load the BioBombe network projection results for Cancer Hallmarks
-
 # In[4]:
 
 
-file = os.path.join("..", "6.biobombe-projection", "results", "tcga",
-                    "gph", "signal", "tcga_z_200_GpH__geneset_scores.tsv.gz")
-
-scores_df = (
-    pd
-    .read_table(file)
-    .query("seed == @seed")
-    .query("z == @z_dim")
+full_coef_id_df = (
+    pd.DataFrame(coef_df.feature.str.split("_").values.tolist(),
+                 columns=['algorithm', 'individual_feature', 'seed', 'k', 'signal'])
 )
 
-scores_df = (
-    scores_df
-    .assign(full_feature=scores_df.algorithm.astype(str) + "_" + scores_df.feature.astype(str),
-            abs_z_score=scores_df.z_score.abs())
-)
-scores_df.head()
+full_coef_id_df = pd.concat([full_coef_id_df, coef_df], axis='columns')
+full_coef_id_df = full_coef_id_df.query("abs > 0").query("signal == 'signal'")
```

Comment:
So if the abs weight is 0 then there is no signal?  This is very liberal, do you not want to look for a stronger signal?

Reply:
These are the coefficients learned in the supervised learning model. I am subseting all features that have some non-zero supervised learning coefficient